workflow/pipeline/photometry.py

```diff
@@ -53,79 +53,262 @@ class EmissionColor(dj.Lookup):
-        # Parameters
+
+        # Parameters (hard-coded, may need to changed later)
         fiber_to_side_mapping = {1: "right", 2: "left"}
         color_mapping = {"g": "green", "r": "red", "b": "blue"}
-        behavior_sample_rate = 200  # original behavioral sampling freq (Hz)
-        target_downsample_rate = 50  # (Hz)
-        downsample_factor = behavior_sample_rate / target_downsample_rate
+        synch_signal_names = ["toBehSys", "fromBehSys"]
+        demod_sample_rate = 600
 
         # Find data dir
-        subject_id = (session.Session & key).fetch1("subject")
         session_dir = (session.SessionDirectory & key).fetch1("session_dir")
         session_full_dir: Path = find_full_path(get_raw_root_data_dir(), session_dir)
-        behavior_dir = session_full_dir / "Behavior"
         photometry_dir = session_full_dir / "Photometry"
 
+        # Read from the meta_info.toml in the photometry folder if exists
+        meta_info_file = list(photometry_dir.glob("*.toml"))[0]
+        meta_info = {}
+        try:
+            with open(meta_info_file.as_posix()) as f:
+                meta_info = tomli.loads(f.read())
+        except FileNotFoundError:
+            logger.info("meta info is missing")
+        light_source_name = meta_info.get("Fiber", {}).get("light_source", "")
+
+        # Read raw data sourced from tdt
         tdt_data: tdt.StructType = tdt.read_block(photometry_dir)
 
-        photometry_df: pd.DataFrame = demodulation.offline_demodulation(
-            tdt_data, z=True, tau=0.05, downsample_fs=600, bandpass_bw=20
+        # Demodulate & downsample raw photometry data
+        # Also returns raw sample rate and list of fibers used
+        photometry_df, fibers, raw_sample_rate = demodulation.offline_demodulation(
+            tdt_data, z=True, tau=0.05, downsample_fs=demod_sample_rate, bandpass_bw=20
         )
-
-        # List working fibers
-        fibers = [
-            i for i in [1, 2] if np.std(tdt_data.streams[f"Fi{i}r"].data[0][5:] > 0.05)
-        ]
         del tdt_data
 
-        channels: T.List[str] = photometry_df.columns.drop(
-            ["toBehSys", "fromBehSys"]
-        ).tolist()
+        # Get trace names e.g., ["detrend_grnR", "raw_grnR"]
+        trace_names: list[str] = photometry_df.columns.drop(synch_signal_names).tolist()
+
+        trace_names = set([name[:-1] for name in trace_names])
+
+        # Store data in this list for ingestion
+        fiber_list: list[dict] = []
+        demodulated_trace_list: list[dict] = []
+
+        # Populate FiberPhotometry
+        beh_synch_signal = photometry_df[synch_signal_names].to_dict("list")
+        beh_synch_signal = {k: np.array(v) for k, v in beh_synch_signal.items()}
+
+        self.insert1(
+            {
+                **key,
+                "light_source_name": light_source_name,
+                "raw_sample_rate": raw_sample_rate,
+                "beh_synch_signal": beh_synch_signal,
+            }
+        )
```

Comment:
If it makes no difference, I'd suggest moving this `self.insert1` further down to be together with the insertions into the part tables. 
The idea is to communicate roughly 3 steps in a typical `make()`: i) fetch the data from upstream, ii) do some computation, iii) insert